lib/server.js

```diff
@@ -86,66 +87,130 @@ Server.prototype._assertHaveAllTasks = function () {
   return Promise.resolve(this.hermes.queues).bind(this)
     .each(function (queueName) {
       if (!this._tasks[queueName]) {
-        throw new Error(queueName + ' handler not defined');
+        throw new Error('ponos.Server: ' + queueName + ' handler not defined');
       }
     });
 };
 
 /**
+ * Helper function that subscribes to a given queue name.
+ * @param {string} queueName Name of the queue.
+ * @return {promise} A promise that resolves post worker creation
+ *   and subscription
+ */
+Server.prototype._subscribe = function (queueName) {
+  this.log.trace('ponos.Server: subscribing to ' + queueName);
+  return this.hermes.subscribeAsync(queueName, function (job, done) {
+    this._runWorker(queueName, job, done);
+  }.bind(this));
+};
+
+/**
+ * Helper function to subscribe to all queues.
+ * @return {promise} Resolved when queues are all subscribed.
+ */
+Server.prototype._subscribeAll = function () {
+  return Promise.resolve(this.hermes.queues).bind(this)
+    .map(this._subscribe);
+};
+
+/**
+ * Runs a worker for the given queue name, job, and acknowledgement callback.
+ * @param {string} queueName Name of the queue.
+ * @param {object} job Job for the worker to perform.
+ * @param {function} done RabbitMQ acknowledgement callback.
+ */
+Server.prototype._runWorker = function (queueName, job, done) {
+  var opts = {};
+  set(opts, this._workerOptions[queueName]);
```

Comment:
this set is looking a lot like a clone to me, is that right? Can I suggest doing one of the below? It just took me way too long to figure out that we're just functionally cloning something...

```js
var opts = assign({}, this._workerOptions[queueName]); // 101/assign, but is more mysterious
// or
var opts = clone(this._workerOptions[queueName]); // 101/clone, is super clear
```